pkg/secrets/secrets.go

```diff
@@ -214,24 +214,43 @@ func (sd *SecretDefinition) addSecrets(client *api.Client, secretResult *SecretR
 
 	// Return last error encountered during processing, if any
 	var lastErr error
+	var value []byte
 
 	singleValueKey := os.Getenv(secretValueKeyPrefix + sd.secretID)
 	if singleValueKey != "" && !sd.plural {
 		v, ok := secretData[singleValueKey]
 		if ok {
-			sd.secrets[singleValueKey] = v.(string)
-			log.Info().Str("key", secretValueKeyPrefix+sd.secretID).Str("value", singleValueKey).Msg("Found an explicit vault value key, will only read value")
-			return nil
+			secretValue, ok := v.(string)
```

Comment:
In the event we might want to support additional types in the future, how about something like this that can be used in both places?

```
func valueConverter(value interface{}) (string, error) {
    switch v.(type) {
    case string:
        return v.(string), nil
    case map[string]interface{}:
        val, err := jason.Marshal(v)
        if err != nil {
            return "", err
        }
        return string(val), nil
    default:
        return "", fmt.Errorf("unsupported value type retrieved from vault: %T", v)
}
```

```
v, ok := secretData[singleValueKey]
if ok { 
    value, err := valueConverter(v)
    if err != nil {
        return fmt.Errorf("failed to convert %v: %w", singleValueKey, err)
    }
    sd.secrets[singleValueKey] = value
}
```

and

```
for k, v := range secretData {
    value, err := valueConverter(v)
    if err != nil {
        return fmt.Errorf("failed to convert %v: %w", k, err)
    }
    switch k { 
    case defaultKeyName:
        sd.secrets[keyName] = value
    default:
        expandedKeyName := fmt.Sprintf("%s_%s", keyName, k)
        sd.secrets[expandedKeyName] = value
    }   
}   
```

Reply:
I addressed most of the comment in the last update. I had a doubt about the `lastError`, I should not use it and just return ?